### PR TITLE
Nullify the addon guid on soft deletion (bug 1176697)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1185789](https://bugzilla.mozilla.org/show_bug.cgi?id=1176697)

### CODE REVIEW

**Comment by kmaglione** [comment]
Can you add a log entry with `keep = True`, and the `guid` and `id` of the add-on before you delete it? Or otherwise store the original guid in another field in the addons table.

We should probably also run a migration to null the IDs of existing deleted add-ons (and ideally add the same log entries).

Looks good otherwise (aside from the failing test, that is). r+wc

**Comment by magopian** [comment]
Updated ;)

File: migrations/818-deleted-addons-nullify-guid.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+This migration empties all the previously soft deleted add-ons."""
+
+import amo
+from addons.models import Addon
+
+
+addons = Addon.unfiltered.filter(status=amo.STATUS_DELETED)
+for addon in addons:
+    amo.log(amo.LOG.DELETE_ADDON, addon.pk, addon.guid, addon)
```
Comment (kmaglione) [inline migrations/818-deleted-addons-nullify-guid.py]: Two minor nits:

1) Can you add `created=addon.modified`, just so we have some consistency between old entries and new?
2) We should probably add a comment to the soft deletion code, that if we stop using soft deletion, we need to make sure these entries aren't cascade deleted.

**Comment by kmaglione** [comment]
Looks good. Thanks! r+wc

**Comment by magopian** [comment]
Added the comment about cascade deletion [here](https://github.com/mozilla/olympia/pull/612/files#diff-49660cbf19988472d86e685e44b83b58R450), is that what you were thinking about?

**Comment by kmaglione** [comment]
Yeah, that's what I meant. (Sorry, I missed that question.)

**Pull request merged**